lib/gems/pending/util/mount/miq_generic_mount_session.rb

```diff
@@ -187,6 +187,14 @@ def reconnect!
     connect
   end
 
+  def self.supports_objects?
```

Comment:
I thought we decided that we didn't need these methods and would use logic more like [this](https://github.com/ManageIQ/manageiq/pull/17689/files#diff-2824c34710b68cc8ae0f96cec49c2a2aR122)?

Reply:
I'm not sure about *we* deciding that... :-)  You made that suggestion this morning on the other PR I believe.  My issue with that is we will definitely also need it for the Openstack Swift subclass as well.  And whatever other object-store based subclass comes up.  So that means checking on a class-by-class basis, which gets ugly.  I much prefer having a class method that controls the functionality.

Reply:
I guess I'm not sure what's stopping us from treating all the mount sessions the same. Since they all respond to `#add` I feel like we should be able to use that method no matter what subclass we're dealing with. That's the point of this kind of duck-typing right?

Reply:
So my take on it is this - I'm not imposing any functionality on NFS or SMB or FTP or whatever other non-object-store based session we might have here.  There is an existing MiqGenericMountSession add method that does a bunch of stuff already.  I'm not sure what calls that but it certainly wasn't previously called during the backup session by the EvmDatabaseOps class.  If I don't test for an object-store based subclass in some way or another every backup for NFS and SMB, etc will invoke that add method.  They shouldn't.

Reply:
> every backup for NFS and SMB, etc will invoke that add method. They shouldn't.

I'm not sure that's true. After a quick look it seems like the `MiqGenericMountSession#add` method puts the file in the specified path relative to the mount point.

In the backup code it looks like we're doing that incidentally by virtue of being able to determine the mount point as well as the destination path. It feels like using the `#add` method would be a no-op for the other mount session types and might actually simplify adding new backup targets like FTP.

Additionally, I think not having to know what type of session you're using will really clean up the implementation here so I think it would be an overall win.

Reply:
Ok and for the restore situation - which is the next PR which I haven't even had the chance to push out yet - EvmDatabaseOps will call the session download method - same issue - can we do that on non-object-store session types?

Reply:
I would hope so. That would make sense to me.

Reply:
Just so everyone reading this knows - making these changes will require going back to manageiq-gems-pending and modifying the add (and probably download) methods for the MiqGenericMountSession class.

Reply:
In what way? What changes are necessary?

Reply:
Aren't we overriding that completely in the s3 mount session class?